linear_algebra/src/rayleigh_quotient.py

```diff
@@ -0,0 +1,65 @@
+"""
+https://en.wikipedia.org/wiki/Rayleigh_quotient
+"""
+import numpy as np
+
+
+def is_hermitian(matrix) -> bool:
+    """
+    Checks if a matrix is Hermitian.
+
+    >>> import numpy as np
+    >>> A = np.matrix([
+    ... [2,    2+1j, 4],
+    ... [2-1j,  3,  1j],
+    ... [4,    -1j,  1]])
+    >>> is_hermitian(A)
+    True
+    >>> A = np.matrix([
+    ... [2,    2+1j, 4+1j],
+    ... [2-1j,  3,  1j],
+    ... [4,    -1j,  1]])
+    >>> is_hermitian(A)
+    False
+    """
+    return np.array_equal(matrix, matrix.H)
+
+
+def rayleigh_quotient(A, v) -> float:
+    """
+    Returns the Rayleigh quotient of a Hermitian matrix A and
+    vector v.
+    >>> import numpy as np
+    >>> A = np.matrix([
+    ... [1,  2, 4],
+    ... [2,  3,  -1],
+    ... [4, -1,  1]
+    ... ])
+    >>> v = np.matrix([
+    ... [1],
+    ... [2],
+    ... [3]
+    ... ])
+    >>> rayleigh_quotient(A, v)
+    matrix([[3.]])
```

Comment:
Would this test pass if v was a list of lists instead of a np.matrix?
`v = [[1], [2], [3]]`

Reply:
No, it wouldn't. Added static typing to these as well.